src/NServiceBus.Newtonsoft.Json/NewtonsoftSerializer.cs

```diff
@@ -22,9 +26,8 @@ public class NewtonsoftSerializer : SerializationDefinition
                 var writerCreator = settings.GetWriterCreator();
                 var serializerSettings = settings.GetSettings();
                 var contentTypeKey = settings.GetContentTypeKey();
-                return new JsonMessageSerializer(mapper, readerCreator, writerCreator, serializerSettings, contentTypeKey);
+                return new JsonMessageSerializer(mapper, readerCreator, writerCreator, serializerSettings, contentTypeKey, global::Newtonsoft.Json.TypeNameHandling.Auto);
```

Comment:
Once we have hit the version of `TreatAsErrorFromVersion` we can replace the implementation with `throw new NotImplementedException();`

We might want to cherry pick this change back to the `release-2.2` branch first because we'll need the implementation there.

Reply:
I have replaced all implementations with `NotImplementedException();`
We will manually add the relevant code to 2.3 instead of cherry picking.